swift_browser_ui_frontend/src/common/store.js

```diff
@@ -145,78 +142,90 @@ const store = new Vuex.Store({
     },
   },
   actions: {
-    updateContainers: async function ({ commit, dispatch }, signal) {
+    updateContainers: async function (
+      { commit, dispatch, state },
+      {route, signal},
+    ) {
       commit("loading", true);
-      let containers = [];
-      await getBuckets().then((ret) => {
-        if (ret.status != 200) {
+      let project = route.params.project;
+      commit("updateContainers", []);
+      let containers;
+      let marker = "";
+      do {
+        containers = [];
+        containers = await getBuckets(
+          project,
+          marker,
+        ).catch(() => {
           commit("loading", false);
-        }
-        containers = ret;
-        commit("updateContainers", ret);
-        commit("loading", false);
-      }).catch(() => {
+        });
         commit("loading", false);
+        if (containers.length > 0) {
+          commit("updateContainers", state.containerCache.concat(containers));
+          marker = containers.pop().name;
+        }
+      } while (containers.length > 0);
+      containers = state.containerCache;
+      dispatch("updateContainerTags", {
+        project,
+        containers,
+        signal,
       });
-      dispatch("updateContainerTags", {containers, signal});
       return containers;
     },
-    updateContainerTags: function ({ commit }, {containers, signal}) {
-      containers.map(async container => {
-        const tags = await getTagsForContainer(container.name, signal);
-        commit(
-          "updateContainerTags", 
-          {containerName: container.name, tags},
-        );
-      });
+    updateContainerTags: function (
+      { commit },
+      {project, containers, signal},
+    ) {
+      if (containers.length < 10000) {
+        console.log("container tag update called.");
+        containers.map(async container => {
+          console.log("Updating tags for container ", container.name);
+          const tags = await getTagsForContainer(
+            project, container.name, signal,
+          );
+          commit(
+            "updateContainerTags", 
+            {containerName: container.name, tags},
+          );
+        });
+      }
     },
     updateObjects: async function (
-      { commit, dispatch, state }, 
-      {route, signal},
+      { commit, dispatch, state },
+      { route, signal },
     ) {
-      let container = route.params.container;
       commit("loading", true);
-      if (route.name == "SharedObjects") {
```

Comment:
What happened to the shared objects?

Reply:
Deprecated, since the object listing  is project aware now we no longer need a different endpoint for fetching over project borders.